drivers/serial/CMakeLists.txt

```diff
@@ -29,6 +29,7 @@ zephyr_library_sources_if_kconfig(uart_psoc6.c)
 zephyr_library_sources_if_kconfig(uart_pl011.c)
 zephyr_library_sources_if_kconfig(uart_rv32m1_lpuart.c)
 zephyr_library_sources_if_kconfig(uart_liteuart.c)
+zephyr_library_sources_if_kconfig(uart_rtt.c)
```

Comment:
By changing this to `zephyr_library_sources_ifdef(CONFIG_UART_RTT_DRIVER uart_rtt.c)` you could make this file being included in the compilation only when this is actually needed (and then you would not need this `#if CONFIG_UART_RTT_DRIVER` in `uart_rtt.c`).

Reply:
That's a good point. I'll fix it.